packages/ingest/test/test-S3ProviderClient.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const fs = require('fs');
+const { basename } = require('path');
+const test = require('ava');
+const S3 = require('@cumulus/aws-client/S3');
+const { randomString } = require('@cumulus/common/test-utils');
+const S3ProviderClient = require('../S3ProviderClient');
+
+const localPath = './tmp.json';
+
+test.before(async (t) => {
+  t.context.sourceBucket = randomString();
+  t.context.sourcePrefix = randomString();
+  t.context.sourceKey = `${t.context.sourcePrefix}/${randomString()}`;
+  t.context.targetBucket = randomString();
+  t.context.fileContent = JSON.stringify({ type: 'fake-test-object' });
+
+  await Promise.all([
+    S3.createBucket(t.context.sourceBucket),
+    S3.createBucket(t.context.targetBucket)
+  ]);
+
+  await S3.s3PutObject({
+    Bucket: t.context.sourceBucket,
+    Key: t.context.sourceKey,
+    Body: t.context.fileContent
+  });
+});
+
+test.after.always(async (t) => {
+  fs.unlinkSync(localPath);
+  return Promise.all([
+    S3.recursivelyDeleteS3Bucket(t.context.sourceBucket),
+    S3.recursivelyDeleteS3Bucket(t.context.targetBucket)
+  ]);
+});
+
+test('S3ProviderClient constructor throws error if no bucket specified', (t) => {
+  t.throws(
+    () => new S3ProviderClient(),
+    'bucket is required'
+  );
+  t.throws(
+    () => new S3ProviderClient({}),
+    'bucket is required'
+  );
+});
+
+test.serial('S3ProviderClient.list lists objects under a path in a bucket', async (t) => {
```

Comment:
Do we care about testing the case of listing objects at the root of the bucket?

Reply:
I guess I can add that.